packages/sources/src/api/createSource.js

```diff
@@ -78,6 +79,11 @@ export const doCreateSource = async (formData, sourceTypes) => {
 
         const [ endpointDataOut, applicationDataOut ] = await Promise.all(promises);
 
+        if (formData.source_type === 'azure') {
```

Comment:
I would remove the billing source endpoint completely and this condition changed to

 `formData.billing_source || formData.credentials`